recipes/gtest/all/conanfile.py

```diff
@@ -101,10 +104,10 @@ def loose_lt_semver(v1, v2):
             min_length = min(len(lv1), len(lv2))
             return lv1[:min_length] < lv2[:min_length]
 
-        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
-        if min_version and loose_lt_semver(str(self.settings.compiler.version), min_version):
+        min_version = self._minimum_compilers_version.get(str(compiler))
+        if min_version and loose_lt_semver(str(compiler.version), min_version):
             raise ConanInvalidConfiguration(
-                f"{self.name} requires {self.settings.compiler} {min_version}. The current compiler is {self.settings.compiler} {self.settings.compiler.version}."
+                f"{self.name} requires {compiler} {min_version}. The current compiler is {compiler} {compiler.version}."
```

Comment:
just a complete name

```suggestion
                f"{self.ref} requires {compiler} {min_version}. The current compiler is {compiler} {compiler.version}."
```